### ISSUE TITLE
Provide Context to scheduler plugins

### ISSUE THREAD

**Issue by zwpaper** [kubernetes-sigs/scheduler-plugins#664]
The general idea would be use context wherever you can:

1. the chained calls in PreFitler/Filter can use the context that is provided by scheduler framework
2. for calls outside PreFilter/Filter/..., we should instantiate one at the start of main() (or make some change in scheduler framework to expose it), and pass it to all plugins

_Originally posted by @Huang-Wei in https://github.com/kubernetes-sigs/scheduler-plugins/pull/655#discussion_r1393097718_
            

/kind enhancement

**Comment by lianghao208** [comment]
To enhance the scheduler framework, do we need to introduce a new parameter into PluginFactory function in k/k framework/runtime/registry.go? so that we can instantiate each plugin with one consolidate context.

**Comment by Huang-Wei** [comment]
> To enhance the scheduler framework, do we need to introduce a new parameter into PluginFactory function in k/k framework/runtime/registry.go?

Yes, that would be an ideal way. I noticed this has been implemented - as a by-product of [116884](https://github.com/kubernetes/kubernetes/pull/116884) (in v1.29)

**Comment by Huang-Wei** [comment]
Changing milestone to v1.29.

/milestone v1.29

**Comment by zwpaper** [comment]
let me have a look at it recently 

/remove-lifecycle stale

**Comment by han-so1omon** [comment]
new contributor. gonna take a look as well. @zwpaper if you already have a fix let me know

**Comment by googs1025** [comment]
Do we have any good suggestions to solve this issue?  
I can try this.
/assign
